matchzoo/preprocessor/process_units.py

```diff
@@ -410,3 +411,108 @@ def transform(self, tokens: list) -> list:
                              'pad mode.'.format(self._pad_mode))
 
         return fixed_tokens.tolist()
+
+
+class EmbeddingUnit(StatefulProcessorUnit):
+    """
+    Embedding class.
+
+    Examples:
+        >>> embed = EmbeddingUnit('tests/unit_test/data/embed_10.txt')
+        >>> # Need term_index
+        >>> embed.fit({'G': 1, 'C': 2, 'D': 3, 'A': 4, '[PAD]': 0})
+        >>> index_state = embed.state['index_state']
+        >>> index_state # doctest: +SKIP
+        {4: 1, 2: 1, 3: 1, 1: 2, 0: 0}
+        >>> embed_mat = embed.state['embed_mat']
+        >>> embed_mat # doctest: +SKIP
+        array([[ 0.       ,  0.       ,  0.       ,  0.       ,  0.        ,
+             0.        ,  0.        ,  0.        ,  0.        ,  0.        ],
+           [ 0.22785383, -0.09405118,  0.15446669,  0.20727136,  0.05943427,
+             0.12673594,  0.02788511, -0.0433806 , -0.20548974,  0.24532762],
+           [ 0.1       ,  0.2       ,  0.3       ,  0.4       ,  0.5       ,
+             0.6       ,  0.7       ,  0.8       ,  0.9       ,  1.        ],
+           [ 0.1       ,  0.2       ,  0.3       ,  0.4       ,  0.5       ,
+             0.6       ,  0.7       ,  0.8       ,  0.9       ,  1.        ],
+           [ 0.1       ,  0.2       ,  0.3       ,  0.4       ,  0.5       ,
+             0.6       ,  0.7       ,  0.8       ,  0.9       ,  1.        ]])
+        >>> embed_mat[0] # doctest: +SKIP
+        array([0., 0., 0., 0., 0., 0., 0., 0., 0., 0.])
+
+    """
+
+    def __init__(self, embed_file: str):
+        """
+        Class initialization.
+
+        :param embed_file: the embeding input file path.
+        """
+        self._state = {}
+        self._embed_file = embed_file
+        self._embed_dim = 0
+        self._index_state = {}
+        self._embed_mat = None
+
+    def fit(self, term_index: dict):
+        """Build a :np.array:`embed_mat` and a :dict:`index_state`."""
+        n_word_in_dict = max(term_index.values()) + 1
+        if n_word_in_dict != len(term_index):
+            print("[Warning] Some words are not shown in term_index({})."
+                  "Total number of words are {}.".format(len(term_index),
+                                                         n_word_in_dict))
+        n_word_share = 0
+
+        embed_file_ptr = open(self._embed_file, 'rb')
```

Comment:
same, what is `ptr`, and do not forget to close. Use context manager `with open`

Reply:
It means embed file pointer.